pkg/kubernetes/node/v1alpha1/node.go

```diff
@@ -135,6 +140,29 @@ func (b *ListBuilder) WithFilter(pred ...Predicate) *ListBuilder {
 	return b
 }
 
+// WithTaints add taints to the Node resource
+func (b *Builder) WithTaints(taintsToAdd []corev1.Taint) *Builder {
+	newTaints := append([]corev1.Taint{}, taintsToAdd...)
+	oldTaints := b.Node.object.Spec.Taints
+
+	for _, oldTaint := range oldTaints {
+		oldTaint := oldTaint
+		existsInNew := false
+		for _, taint := range newTaints {
```

Comment:
Will it be good to have a type & Contains method
```go
type TaintList []corev1.Taint

func (l TaintList) Contains(given corev1.Taint) bool {
  for _, available := range l {
    if available.MatchTaint(&given) {
      return true
    }
  }
  return false
}
```

Reply:
We will be refactoring taints code in upoming PR .. if that ok ?

Reply:
Can we make use of above snippet?
We need to add UT as well.
Let us do both the above in this PR.